### ISSUE TITLE
When drinking use and quickly switching ithe item slots to another, both juices get consumed and the player recieves a singular glass blottle

### ISSUE THREAD

**Issue by marvjosh1** [Slimefun/Slimefun4#3264]
### ❗ Checklist

- [X] I am using the official english version of Slimefun and did not modify the jar.
- [X] I am using an up to date "DEV" (not "RC") version of Slimefun.
- [X] I am aware that issues related to Slimefun addons need to be reported on their bug trackers and not here.
- [X] I searched for similar open issues and could not find an existing bug report on this.

### 📍 Description

When drinking a juice and in the last instance of drinking it switching to another hotbar slot with a juice, bot get consumed and the player recieves a glass bottle.

### 📑 Reproduction Steps

1. Place 2 juices in your inventory
2. Start drinking one of them
3. The last minute before the player is finishing drinking it, switch to the hotbar slot with the other juice

### 💡 Expected Behavior

The expected behaviour is that only one of the juices would be consumed, and the player doesnt get a glass bottle from it.

### 📷 Screenshots / Videos

https://www.youtube.com/watch?v=GE1DBlG2I84

### 💻 Server Software

Purpur

### 🎮 Minecraft Version

1.17.x

### ⭐ Slimefun version

https://i.imgur.com/Hgp4T9g.png
https://i.imgur.com/AuIoUYW.png

**Comment by variananora** [comment]
Yep, can confirm this. But the timing is quite hard for me to get it right lol.
https://youtu.be/sbAQN17qy68

**Comment by TweepCoding** [comment]
Looking at the code, it seems that this is caused by a timing issue due to Slimefun.runSync(), in which, when well timed, the code can enter a race in this form:

* Juice handler verifies interaction, applies effect and goes to remove bottle
* The function then calls runSync() to schedule the removal of the item in hand.
* While the action is still being scheduled, the player switches exactly at that tick to another spot in the hotbar.
* Minecraft does his thing with bottles, and replaces the old slot with the juice with an empty bottle.
* runSync() activates, and erases item in hand.

This would explain why the timing would need to be very precise.
This is all pure hypothesis though, I will do actual testing tomorrow to see if this is the case.
